sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -299,6 +305,8 @@ public void Start()
                 }
 
                 _diagnosticSource.Write(_activityName + ".Start", _diagnosticSourceArgs ?? _currentActivity);
+
+                return _currentActivity;
```

Comment:
Is it safe to return `Activity` instance here?

Reply:
yes, it's safe. Anyone can also get it with `Activity.Current`, but we're making a small optimization. Also, `ActivityAdapter` (where this method is defined) is a private class and only used in `DiagnosticScope`, so we fully control who/how will call it